src/state/api/utils.ts

```diff
@@ -32,36 +31,38 @@ export const generateTidligsteHelUttaksalderRequestBody = (args: {
   }
 }
 
-// TODO denne skal aktiveres ifbm tidligste uttaksalder for avansert fane
-// export const generateTidligsteGradertUttaksalderRequestBody = (args: {
-//   afp: AfpRadio | null
-//   sivilstand?: Sivilstand | null | undefined
-//   harSamboer: boolean | null
-//   aarligInntektFoerUttakBeloep: number
-//   gradertUttak: Omit<GradertUttaksperiode, 'uttaksperiode'>
-// }): TidligsteGradertUttaksalderRequestBody | undefined => {
-//   const {
-//     afp,
-//     sivilstand,
-//     harSamboer,
-//     aarligInntektFoerUttakBeloep,
-//     gradertUttak,
-//   } = args
+export const generateTidligsteGradertUttaksalderRequestBody = (args: {
+  afp: AfpRadio | null
+  sivilstand?: Sivilstand | null | undefined
+  harSamboer: boolean | null
+  aarligInntektFoerUttakBeloep: number
+  gradertUttak: Omit<GradertUttaksperiode, 'uttaksalder'>
+  heltUttak: HeltUttaksperiode
+}): TidligsteGradertUttaksalderRequestBody | undefined => {
+  const {
+    afp,
+    sivilstand,
+    harSamboer,
+    aarligInntektFoerUttakBeloep,
+    gradertUttak,
+    heltUttak,
+  } = args
 
-//   return {
-//     simuleringstype:
-//       afp === 'ja_privat' ? 'ALDERSPENSJON_MED_AFP_PRIVAT' : 'ALDERSPENSJON',
-//     harEps: harSamboer !== null ? harSamboer : false, // TODO sjekke med Espen om det er riktig
-//     aarligInntekt: aarligInntektFoerUttakBeloep,
-//     sivilstand:
-//       sivilstand && checkHarSamboer(sivilstand)
-//         ? sivilstand
-//         : harSamboer
-//           ? 'SAMBOER'
-//           : 'UGIFT',
-//     gradertUttak,
-//   }
-// }
+  return {
+    simuleringstype:
+      afp === 'ja_privat' ? 'ALDERSPENSJON_MED_AFP_PRIVAT' : 'ALDERSPENSJON',
+    harEps: harSamboer !== null ? harSamboer : false, // TODO sjekke med Espen om det er riktig
```

Comment:
Her kan ev. frontend la være å sende med `harEps`.
Da vil backend utlede `harEps` basert på sivilstand, basert på [backend: Sivilstand](https://github.com/navikt/pensjonskalkulator-backend/blob/main/src/main/kotlin/no/nav/pensjon/kalkulator/person/Sivilstand.kt)

Reply:
Her tror jeg at frontend sitter med data av bedre kvalitet som kan være lur å bruke.
På steg 4 av stegvisningen, og dersom brukeren har en annen sivilstand enn 'GIFT' eller 'REGISTRERT_PARTNER' så ber vi brukeren eksplisitt om å svare på om hen har samboer.
Dvs. at vi har informasjon om samboerskap som kommer fra brukeren (og ikke bare basert på sivilstand).
Sjekk for null er bare en guard, men på beregningssiden kan ikke dette være null. (fordi brukeren er omdirigert hvis /person kallet skulle ha feilet)

Hva tenker du?

Reply:
Ja, godt poeng at brukeren angir `harSamboer`.
Men hvis brukeren ikke angir dette, så må `harEps` utledes fra sivilstand (og da kan enten frontend utlede det selv, eller sende `null` til backend, som da vil gjøre det).
Så jeg tror det blir feil å sende `false` dersom `harSamboer === null`.

Reply:
Ok. Den vil nok aldri være `null` på det tidspunktet, men for at det skal være korrekt så er det best at frontend sender `null` videre og ikke false. Jeg retter opp :)